{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib\n",
    "matplotlib.rcParams['pdf.fonttype'] = 42\n",
    "matplotlib.rcParams['ps.fonttype'] = 42"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "hamsini_table = pd.read_csv('/data/passala/Collaborator_Data/Zach_Pan_solanum_Data/Hamsini_generated_tables/tomato_paralogs_Sri_CompDrift.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene1</th>\n",
       "      <th>Gene2</th>\n",
       "      <th>OGs</th>\n",
       "      <th>coexp_atlas</th>\n",
       "      <th>coexp_tissue</th>\n",
       "      <th>FC_atlas</th>\n",
       "      <th>spec_atlas</th>\n",
       "      <th>FC_tissue</th>\n",
       "      <th>spec_tissue</th>\n",
       "      <th>numtissue_atlas</th>\n",
       "      <th>...</th>\n",
       "      <th>log2FC_min_atlas</th>\n",
       "      <th>log2FC_max_atlas</th>\n",
       "      <th>log2FC_mean_atlas</th>\n",
       "      <th>log2FC_sd_atlas</th>\n",
       "      <th>numtissue_tissue</th>\n",
       "      <th>new_coexp_tissue</th>\n",
       "      <th>log2FC_min_tissue</th>\n",
       "      <th>log2FC_max_tissue</th>\n",
       "      <th>log2FC_mean_tissue</th>\n",
       "      <th>log2FC_sd_tissue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Solyc01g014055</td>\n",
       "      <td>Solyc01g056575</td>\n",
       "      <td>OG0000006</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Solyc01g014055</td>\n",
       "      <td>Solyc01g058003</td>\n",
       "      <td>OG0000006</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Solyc01g014055</td>\n",
       "      <td>Solyc01g066055</td>\n",
       "      <td>OG0000006</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Solyc01g014055</td>\n",
       "      <td>Solyc04g045425</td>\n",
       "      <td>OG0000006</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Solyc01g014055</td>\n",
       "      <td>Solyc04g050745</td>\n",
       "      <td>OG0000006</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25342</th>\n",
       "      <td>Solyc12g049611</td>\n",
       "      <td>Solyc12g049616</td>\n",
       "      <td>OG0036202</td>\n",
       "      <td>0.856193</td>\n",
       "      <td>0.631959</td>\n",
       "      <td>0.763265</td>\n",
       "      <td>0.821843</td>\n",
       "      <td>0.554708</td>\n",
       "      <td>0.679172</td>\n",
       "      <td>12.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.388222</td>\n",
       "      <td>3.339848</td>\n",
       "      <td>1.047086</td>\n",
       "      <td>1.478092</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.415037</td>\n",
       "      <td>-0.415037</td>\n",
       "      <td>-0.415037</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25343</th>\n",
       "      <td>Solyc12g049614</td>\n",
       "      <td>Solyc12g049616</td>\n",
       "      <td>OG0036202</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25344</th>\n",
       "      <td>Solyc01g103310</td>\n",
       "      <td>Solyc01g150153</td>\n",
       "      <td>OG0038467</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25345</th>\n",
       "      <td>Solyc02g088950</td>\n",
       "      <td>Solyc02g088980</td>\n",
       "      <td>OG0038488</td>\n",
       "      <td>0.855671</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.927567</td>\n",
       "      <td>0.961823</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.336638</td>\n",
       "      <td>-1.657256</td>\n",
       "      <td>-1.996947</td>\n",
       "      <td>0.480396</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25346</th>\n",
       "      <td>Solyc08g044300</td>\n",
       "      <td>Solyc08g150123</td>\n",
       "      <td>OG0038616</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>25347 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                Gene1           Gene2        OGs  coexp_atlas  coexp_tissue  \\\n",
       "0      Solyc01g014055  Solyc01g056575  OG0000006          NaN           NaN   \n",
       "1      Solyc01g014055  Solyc01g058003  OG0000006          NaN           NaN   \n",
       "2      Solyc01g014055  Solyc01g066055  OG0000006          NaN           NaN   \n",
       "3      Solyc01g014055  Solyc04g045425  OG0000006          NaN           NaN   \n",
       "4      Solyc01g014055  Solyc04g050745  OG0000006          NaN           NaN   \n",
       "...               ...             ...        ...          ...           ...   \n",
       "25342  Solyc12g049611  Solyc12g049616  OG0036202     0.856193      0.631959   \n",
       "25343  Solyc12g049614  Solyc12g049616  OG0036202          NaN           NaN   \n",
       "25344  Solyc01g103310  Solyc01g150153  OG0038467          NaN           NaN   \n",
       "25345  Solyc02g088950  Solyc02g088980  OG0038488     0.855671           NaN   \n",
       "25346  Solyc08g044300  Solyc08g150123  OG0038616          NaN           NaN   \n",
       "\n",
       "       FC_atlas  spec_atlas  FC_tissue  spec_tissue  numtissue_atlas  ...  \\\n",
       "0           NaN         NaN        NaN          NaN              NaN  ...   \n",
       "1           NaN         NaN        NaN          NaN              NaN  ...   \n",
       "2           NaN         NaN        NaN          NaN              NaN  ...   \n",
       "3           NaN         NaN        NaN          NaN              NaN  ...   \n",
       "4           NaN         NaN        NaN          NaN              NaN  ...   \n",
       "...         ...         ...        ...          ...              ...  ...   \n",
       "25342  0.763265    0.821843   0.554708     0.679172             12.0  ...   \n",
       "25343       NaN         NaN        NaN          NaN              NaN  ...   \n",
       "25344       NaN         NaN        NaN          NaN              NaN  ...   \n",
       "25345  0.927567    0.961823        NaN          NaN              2.0  ...   \n",
       "25346       NaN         NaN        NaN          NaN              NaN  ...   \n",
       "\n",
       "       log2FC_min_atlas  log2FC_max_atlas  log2FC_mean_atlas  log2FC_sd_atlas  \\\n",
       "0                   NaN               NaN                NaN              NaN   \n",
       "1                   NaN               NaN                NaN              NaN   \n",
       "2                   NaN               NaN                NaN              NaN   \n",
       "3                   NaN               NaN                NaN              NaN   \n",
       "4                   NaN               NaN                NaN              NaN   \n",
       "...                 ...               ...                ...              ...   \n",
       "25342         -0.388222          3.339848           1.047086         1.478092   \n",
       "25343               NaN               NaN                NaN              NaN   \n",
       "25344               NaN               NaN                NaN              NaN   \n",
       "25345         -2.336638         -1.657256          -1.996947         0.480396   \n",
       "25346               NaN               NaN                NaN              NaN   \n",
       "\n",
       "       numtissue_tissue  new_coexp_tissue  log2FC_min_tissue  \\\n",
       "0                   NaN               NaN                NaN   \n",
       "1                   NaN               NaN                NaN   \n",
       "2                   NaN               NaN                NaN   \n",
       "3                   NaN               NaN                NaN   \n",
       "4                   NaN               NaN                NaN   \n",
       "...                 ...               ...                ...   \n",
       "25342               1.0               NaN          -0.415037   \n",
       "25343               NaN               NaN                NaN   \n",
       "25344               NaN               NaN                NaN   \n",
       "25345               NaN               NaN                NaN   \n",
       "25346               NaN               NaN                NaN   \n",
       "\n",
       "       log2FC_max_tissue  log2FC_mean_tissue  log2FC_sd_tissue  \n",
       "0                    NaN                 NaN               NaN  \n",
       "1                    NaN                 NaN               NaN  \n",
       "2                    NaN                 NaN               NaN  \n",
       "3                    NaN                 NaN               NaN  \n",
       "4                    NaN                 NaN               NaN  \n",
       "...                  ...                 ...               ...  \n",
       "25342          -0.415037           -0.415037               NaN  \n",
       "25343                NaN                 NaN               NaN  \n",
       "25344                NaN                 NaN               NaN  \n",
       "25345                NaN                 NaN               NaN  \n",
       "25346                NaN                 NaN               NaN  \n",
       "\n",
       "[25347 rows x 21 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hamsini_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Gene1', 'Gene2', 'OGs', 'coexp_atlas', 'coexp_tissue', 'FC_atlas',\n",
       "       'spec_atlas', 'FC_tissue', 'spec_tissue', 'numtissue_atlas',\n",
       "       'new_coexp_atlas', 'log2FC_min_atlas', 'log2FC_max_atlas',\n",
       "       'log2FC_mean_atlas', 'log2FC_sd_atlas', 'numtissue_tissue',\n",
       "       'new_coexp_tissue', 'log2FC_min_tissue', 'log2FC_max_tissue',\n",
       "       'log2FC_mean_tissue', 'log2FC_sd_tissue'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hamsini_table.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>numtissue_atlas</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25342</th>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25343</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25344</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25345</th>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25346</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>25347 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       numtissue_atlas\n",
       "0                  NaN\n",
       "1                  NaN\n",
       "2                  NaN\n",
       "3                  NaN\n",
       "4                  NaN\n",
       "...                ...\n",
       "25342             12.0\n",
       "25343              NaN\n",
       "25344              NaN\n",
       "25345              2.0\n",
       "25346              NaN\n",
       "\n",
       "[25347 rows x 1 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hamsini_table[['numtissue_atlas']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "hamsini_shortlist = hamsini_table.loc[(hamsini_table['coexp_atlas'] >.7) &(hamsini_table['numtissue_atlas'] > 3)&(hamsini_table['log2FC_mean_atlas']>1)&(hamsini_table['log2FC_sd_atlas']<.5)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene1</th>\n",
       "      <th>Gene2</th>\n",
       "      <th>OGs</th>\n",
       "      <th>coexp_atlas</th>\n",
       "      <th>coexp_tissue</th>\n",
       "      <th>FC_atlas</th>\n",
       "      <th>spec_atlas</th>\n",
       "      <th>FC_tissue</th>\n",
       "      <th>spec_tissue</th>\n",
       "      <th>numtissue_atlas</th>\n",
       "      <th>...</th>\n",
       "      <th>log2FC_min_atlas</th>\n",
       "      <th>log2FC_max_atlas</th>\n",
       "      <th>log2FC_mean_atlas</th>\n",
       "      <th>log2FC_sd_atlas</th>\n",
       "      <th>numtissue_tissue</th>\n",
       "      <th>new_coexp_tissue</th>\n",
       "      <th>log2FC_min_tissue</th>\n",
       "      <th>log2FC_max_tissue</th>\n",
       "      <th>log2FC_mean_tissue</th>\n",
       "      <th>log2FC_sd_tissue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1508</th>\n",
       "      <td>Solyc04g011390</td>\n",
       "      <td>Solyc05g054610</td>\n",
       "      <td>OG0000026</td>\n",
       "      <td>0.998242</td>\n",
       "      <td>0.997688</td>\n",
       "      <td>0.899756</td>\n",
       "      <td>0.845550</td>\n",
       "      <td>0.998674</td>\n",
       "      <td>0.998742</td>\n",
       "      <td>17.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.645516</td>\n",
       "      <td>3.542327</td>\n",
       "      <td>2.360036</td>\n",
       "      <td>0.445612</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.986632</td>\n",
       "      <td>-1.000000</td>\n",
       "      <td>1.589492</td>\n",
       "      <td>0.850125</td>\n",
       "      <td>0.935544</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1517</th>\n",
       "      <td>Solyc04g011390</td>\n",
       "      <td>Solyc11g072840</td>\n",
       "      <td>OG0000026</td>\n",
       "      <td>0.998675</td>\n",
       "      <td>0.995199</td>\n",
       "      <td>0.899786</td>\n",
       "      <td>0.845638</td>\n",
       "      <td>0.997816</td>\n",
       "      <td>0.997125</td>\n",
       "      <td>16.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.885063</td>\n",
       "      <td>2.489958</td>\n",
       "      <td>1.304373</td>\n",
       "      <td>0.468712</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.977245</td>\n",
       "      <td>3.095924</td>\n",
       "      <td>4.206383</td>\n",
       "      <td>3.592003</td>\n",
       "      <td>0.447899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1543</th>\n",
       "      <td>Solyc06g005430</td>\n",
       "      <td>Solyc11g066160</td>\n",
       "      <td>OG0000026</td>\n",
       "      <td>0.997317</td>\n",
       "      <td>0.985899</td>\n",
       "      <td>0.898173</td>\n",
       "      <td>0.826985</td>\n",
       "      <td>0.994256</td>\n",
       "      <td>0.996720</td>\n",
       "      <td>17.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.510119</td>\n",
       "      <td>4.043955</td>\n",
       "      <td>3.160603</td>\n",
       "      <td>0.424230</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.949132</td>\n",
       "      <td>0.181838</td>\n",
       "      <td>3.514573</td>\n",
       "      <td>1.281371</td>\n",
       "      <td>1.172641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1559</th>\n",
       "      <td>Solyc06g075850</td>\n",
       "      <td>Solyc06g075930</td>\n",
       "      <td>OG0000026</td>\n",
       "      <td>0.998907</td>\n",
       "      <td>0.995509</td>\n",
       "      <td>0.898090</td>\n",
       "      <td>0.748409</td>\n",
       "      <td>0.998243</td>\n",
       "      <td>0.999079</td>\n",
       "      <td>17.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.095467</td>\n",
       "      <td>2.390936</td>\n",
       "      <td>1.816379</td>\n",
       "      <td>0.413969</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.977422</td>\n",
       "      <td>0.459432</td>\n",
       "      <td>4.426265</td>\n",
       "      <td>1.842428</td>\n",
       "      <td>1.202599</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3121</th>\n",
       "      <td>Solyc01g074000</td>\n",
       "      <td>Solyc01g080600</td>\n",
       "      <td>OG0000033</td>\n",
       "      <td>0.997613</td>\n",
       "      <td>0.882802</td>\n",
       "      <td>0.898274</td>\n",
       "      <td>0.796348</td>\n",
       "      <td>0.934411</td>\n",
       "      <td>0.911762</td>\n",
       "      <td>13.0</td>\n",
       "      <td>...</td>\n",
       "      <td>3.471106</td>\n",
       "      <td>4.672119</td>\n",
       "      <td>4.364960</td>\n",
       "      <td>0.335908</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>8.917159</td>\n",
       "      <td>8.986980</td>\n",
       "      <td>8.952069</td>\n",
       "      <td>0.049371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24622</th>\n",
       "      <td>Solyc06g083090</td>\n",
       "      <td>Solyc06g083100</td>\n",
       "      <td>OG0006221</td>\n",
       "      <td>0.959889</td>\n",
       "      <td>0.692153</td>\n",
       "      <td>0.822121</td>\n",
       "      <td>0.844025</td>\n",
       "      <td>0.822639</td>\n",
       "      <td>0.690808</td>\n",
       "      <td>8.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.728475</td>\n",
       "      <td>2.929909</td>\n",
       "      <td>2.042617</td>\n",
       "      <td>0.370602</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.321928</td>\n",
       "      <td>5.321928</td>\n",
       "      <td>5.321928</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24726</th>\n",
       "      <td>Solyc11g008110</td>\n",
       "      <td>Solyc11g020180</td>\n",
       "      <td>OG0008314</td>\n",
       "      <td>0.988990</td>\n",
       "      <td>0.463262</td>\n",
       "      <td>0.947735</td>\n",
       "      <td>0.898457</td>\n",
       "      <td>0.297676</td>\n",
       "      <td>0.148351</td>\n",
       "      <td>7.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.725931</td>\n",
       "      <td>2.383131</td>\n",
       "      <td>2.085396</td>\n",
       "      <td>0.241926</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.000000</td>\n",
       "      <td>-1.000000</td>\n",
       "      <td>-1.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24744</th>\n",
       "      <td>Solyc12g098463</td>\n",
       "      <td>Solyc12g098467</td>\n",
       "      <td>OG0008506</td>\n",
       "      <td>0.939124</td>\n",
       "      <td>0.980108</td>\n",
       "      <td>0.979775</td>\n",
       "      <td>0.996383</td>\n",
       "      <td>0.995865</td>\n",
       "      <td>0.996091</td>\n",
       "      <td>17.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.933543</td>\n",
       "      <td>4.389846</td>\n",
       "      <td>3.871429</td>\n",
       "      <td>0.389279</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.939008</td>\n",
       "      <td>-1.584963</td>\n",
       "      <td>-0.316027</td>\n",
       "      <td>-1.068042</td>\n",
       "      <td>0.337460</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24940</th>\n",
       "      <td>Solyc06g009420</td>\n",
       "      <td>Solyc06g009490</td>\n",
       "      <td>OG0023112</td>\n",
       "      <td>0.990981</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.897916</td>\n",
       "      <td>0.878454</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.547099</td>\n",
       "      <td>3.370331</td>\n",
       "      <td>2.958930</td>\n",
       "      <td>0.330099</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25249</th>\n",
       "      <td>Solyc07g025200</td>\n",
       "      <td>Solyc07g025230</td>\n",
       "      <td>OG0028096</td>\n",
       "      <td>0.996955</td>\n",
       "      <td>0.886494</td>\n",
       "      <td>0.981462</td>\n",
       "      <td>0.971760</td>\n",
       "      <td>0.894018</td>\n",
       "      <td>0.977626</td>\n",
       "      <td>10.0</td>\n",
       "      <td>...</td>\n",
       "      <td>3.375911</td>\n",
       "      <td>3.811815</td>\n",
       "      <td>3.607786</td>\n",
       "      <td>0.161333</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.518771</td>\n",
       "      <td>-0.736966</td>\n",
       "      <td>1.807355</td>\n",
       "      <td>0.738966</td>\n",
       "      <td>0.948493</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>162 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                Gene1           Gene2        OGs  coexp_atlas  coexp_tissue  \\\n",
       "1508   Solyc04g011390  Solyc05g054610  OG0000026     0.998242      0.997688   \n",
       "1517   Solyc04g011390  Solyc11g072840  OG0000026     0.998675      0.995199   \n",
       "1543   Solyc06g005430  Solyc11g066160  OG0000026     0.997317      0.985899   \n",
       "1559   Solyc06g075850  Solyc06g075930  OG0000026     0.998907      0.995509   \n",
       "3121   Solyc01g074000  Solyc01g080600  OG0000033     0.997613      0.882802   \n",
       "...               ...             ...        ...          ...           ...   \n",
       "24622  Solyc06g083090  Solyc06g083100  OG0006221     0.959889      0.692153   \n",
       "24726  Solyc11g008110  Solyc11g020180  OG0008314     0.988990      0.463262   \n",
       "24744  Solyc12g098463  Solyc12g098467  OG0008506     0.939124      0.980108   \n",
       "24940  Solyc06g009420  Solyc06g009490  OG0023112     0.990981           NaN   \n",
       "25249  Solyc07g025200  Solyc07g025230  OG0028096     0.996955      0.886494   \n",
       "\n",
       "       FC_atlas  spec_atlas  FC_tissue  spec_tissue  numtissue_atlas  ...  \\\n",
       "1508   0.899756    0.845550   0.998674     0.998742             17.0  ...   \n",
       "1517   0.899786    0.845638   0.997816     0.997125             16.0  ...   \n",
       "1543   0.898173    0.826985   0.994256     0.996720             17.0  ...   \n",
       "1559   0.898090    0.748409   0.998243     0.999079             17.0  ...   \n",
       "3121   0.898274    0.796348   0.934411     0.911762             13.0  ...   \n",
       "...         ...         ...        ...          ...              ...  ...   \n",
       "24622  0.822121    0.844025   0.822639     0.690808              8.0  ...   \n",
       "24726  0.947735    0.898457   0.297676     0.148351              7.0  ...   \n",
       "24744  0.979775    0.996383   0.995865     0.996091             17.0  ...   \n",
       "24940  0.897916    0.878454        NaN          NaN              6.0  ...   \n",
       "25249  0.981462    0.971760   0.894018     0.977626             10.0  ...   \n",
       "\n",
       "       log2FC_min_atlas  log2FC_max_atlas  log2FC_mean_atlas  log2FC_sd_atlas  \\\n",
       "1508           1.645516          3.542327           2.360036         0.445612   \n",
       "1517           0.885063          2.489958           1.304373         0.468712   \n",
       "1543           2.510119          4.043955           3.160603         0.424230   \n",
       "1559           1.095467          2.390936           1.816379         0.413969   \n",
       "3121           3.471106          4.672119           4.364960         0.335908   \n",
       "...                 ...               ...                ...              ...   \n",
       "24622          1.728475          2.929909           2.042617         0.370602   \n",
       "24726          1.725931          2.383131           2.085396         0.241926   \n",
       "24744          2.933543          4.389846           3.871429         0.389279   \n",
       "24940          2.547099          3.370331           2.958930         0.330099   \n",
       "25249          3.375911          3.811815           3.607786         0.161333   \n",
       "\n",
       "       numtissue_tissue  new_coexp_tissue  log2FC_min_tissue  \\\n",
       "1508               11.0          0.986632          -1.000000   \n",
       "1517                9.0          0.977245           3.095924   \n",
       "1543               11.0          0.949132           0.181838   \n",
       "1559               11.0          0.977422           0.459432   \n",
       "3121                2.0          1.000000           8.917159   \n",
       "...                 ...               ...                ...   \n",
       "24622               1.0               NaN           5.321928   \n",
       "24726               1.0               NaN          -1.000000   \n",
       "24744              12.0          0.939008          -1.584963   \n",
       "24940               NaN               NaN                NaN   \n",
       "25249               6.0          0.518771          -0.736966   \n",
       "\n",
       "       log2FC_max_tissue  log2FC_mean_tissue  log2FC_sd_tissue  \n",
       "1508            1.589492            0.850125          0.935544  \n",
       "1517            4.206383            3.592003          0.447899  \n",
       "1543            3.514573            1.281371          1.172641  \n",
       "1559            4.426265            1.842428          1.202599  \n",
       "3121            8.986980            8.952069          0.049371  \n",
       "...                  ...                 ...               ...  \n",
       "24622           5.321928            5.321928               NaN  \n",
       "24726          -1.000000           -1.000000               NaN  \n",
       "24744          -0.316027           -1.068042          0.337460  \n",
       "24940                NaN                 NaN               NaN  \n",
       "25249           1.807355            0.738966          0.948493  \n",
       "\n",
       "[162 rows x 21 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hamsini_shortlist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "list_of_orthogroups_with_fc_thresholded_tomato = list(set(hamsini_shortlist['OGs']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "orthogroups = pd.read_csv('/data/passala/Collaborator_Data/Zach_Pan_solanum_Data/Orthology_Tables/Orthogroups.tsv', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "ortho_col_name_converter = {'Aethiopicum':'Solaet3_3.0.4','Candidum':'Solcan1_1.0.4','Macrocarpon':'Solmac3_3.0.4','Muricatum 1':'Solmur2hap1_2.0.4','Muricatum 2':'Solmur2hap2_2.0.4','Prinophyllum':'Solpri1_2.0.4','Quitoense':'Solqui2_2.0.4'}\n",
    "ortho_group_dictionary_of_gene_groups = {}\n",
    "for species in ortho_col_name_converter.keys():\n",
    "    current_ortho_df = orthogroups[['Orthogroup',ortho_col_name_converter[species]]]\n",
    "    current_ortho_df = current_ortho_df.dropna()\n",
    "    df_conversion_base = []\n",
    "    list_of_lenghts = []\n",
    "    for Orthogroup_row in current_ortho_df.iterrows():\n",
    "        current_orthogroup = Orthogroup_row[1][0]\n",
    "        current_list_of_genes_in_group = Orthogroup_row[1][1].replace(\" \",'').split(',')\n",
    "        list_of_lenghts.append(len(current_list_of_genes_in_group))\n",
    "        if len(current_list_of_genes_in_group) >30 or len(current_list_of_genes_in_group) == 1:\n",
    "            continue\n",
    "        orthogroup_list = [current_orthogroup]*len(current_list_of_genes_in_group)\n",
    "        zipped_list = zip(orthogroup_list, current_list_of_genes_in_group)\n",
    "        df_conversion_base.extend(list(zipped_list))\n",
    "        final_long_df = pd.DataFrame(data=df_conversion_base, columns = ['Orthogroup','Gene'])\n",
    "        ortho_group_dictionary_of_gene_groups[species]=final_long_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Cannot index with multidimensional key",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[0;32mIn [23], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m ortho_group_dictionary_of_gene_groups[\u001b[39m'\u001b[39m\u001b[39mQuitoense\u001b[39m\u001b[39m'\u001b[39m]\u001b[39m.\u001b[39mloc[ortho_group_dictionary_of_gene_groups[\u001b[39m'\u001b[39m\u001b[39mQuitoense\u001b[39m\u001b[39m'\u001b[39m]]\n",
      "File \u001b[0;32m~/miniconda3/envs/Single_cell_data/lib/python3.9/site-packages/pandas/core/indexing.py:1074\u001b[0m, in \u001b[0;36m_LocationIndexer.__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   1071\u001b[0m axis \u001b[39m=\u001b[39m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39maxis \u001b[39mor\u001b[39;00m \u001b[39m0\u001b[39m\n\u001b[1;32m   1073\u001b[0m maybe_callable \u001b[39m=\u001b[39m com\u001b[39m.\u001b[39mapply_if_callable(key, \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mobj)\n\u001b[0;32m-> 1074\u001b[0m \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_getitem_axis(maybe_callable, axis\u001b[39m=\u001b[39;49maxis)\n",
      "File \u001b[0;32m~/miniconda3/envs/Single_cell_data/lib/python3.9/site-packages/pandas/core/indexing.py:1300\u001b[0m, in \u001b[0;36m_LocIndexer._getitem_axis\u001b[0;34m(self, key, axis)\u001b[0m\n\u001b[1;32m   1297\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mnot\u001b[39;00m (\u001b[39misinstance\u001b[39m(key, \u001b[39mtuple\u001b[39m) \u001b[39mand\u001b[39;00m \u001b[39misinstance\u001b[39m(labels, MultiIndex)):\n\u001b[1;32m   1299\u001b[0m     \u001b[39mif\u001b[39;00m \u001b[39mhasattr\u001b[39m(key, \u001b[39m\"\u001b[39m\u001b[39mndim\u001b[39m\u001b[39m\"\u001b[39m) \u001b[39mand\u001b[39;00m key\u001b[39m.\u001b[39mndim \u001b[39m>\u001b[39m \u001b[39m1\u001b[39m:\n\u001b[0;32m-> 1300\u001b[0m         \u001b[39mraise\u001b[39;00m \u001b[39mValueError\u001b[39;00m(\u001b[39m\"\u001b[39m\u001b[39mCannot index with multidimensional key\u001b[39m\u001b[39m\"\u001b[39m)\n\u001b[1;32m   1302\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_getitem_iterable(key, axis\u001b[39m=\u001b[39maxis)\n\u001b[1;32m   1304\u001b[0m \u001b[39m# nested tuple slicing\u001b[39;00m\n",
      "\u001b[0;31mValueError\u001b[0m: Cannot index with multidimensional key"
     ]
    }
   ],
   "source": [
    "ortho_group_dictionary_of_gene_groups['Quitoense'].loc[ortho_group_dictionary_of_gene_groups['Quitoense']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Orthogroup</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Orthogroup, Gene]\n",
       "Index: []"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "current_list_of_genes.loc[current_list_of_genes['Orthogroup'] == 'test']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "ortho_groups_and_number_of_genes_present_in_each_species = pd.DataFrame(index = list_of_orthogroups_with_fc_thresholded_tomato)\n",
    "for species in ortho_group_dictionary_of_gene_groups.keys():\n",
    "    current_list_of_genes = ortho_group_dictionary_of_gene_groups[species]\n",
    "    all_orthogroups_number_present = []\n",
    "    for orthogroup in list_of_orthogroups_with_fc_thresholded_tomato:\n",
    "        genes_in_orthogroup = current_list_of_genes.loc[current_list_of_genes['Orthogroup'] == orthogroup]\n",
    "        numb_of_that_gene_in_species = len(genes_in_orthogroup)\n",
    "        all_orthogroups_number_present.append(numb_of_that_gene_in_species)\n",
    "    ortho_groups_and_number_of_genes_present_in_each_species[species] = all_orthogroups_number_present\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Aethiopicum</th>\n",
       "      <th>Candidum</th>\n",
       "      <th>Macrocarpon</th>\n",
       "      <th>Muricatum 1</th>\n",
       "      <th>Muricatum 2</th>\n",
       "      <th>Prinophyllum</th>\n",
       "      <th>Quitoense</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>OG0002827</th>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0005089</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0001699</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0000192</th>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0000040</th>\n",
       "      <td>10</td>\n",
       "      <td>14</td>\n",
       "      <td>16</td>\n",
       "      <td>13</td>\n",
       "      <td>9</td>\n",
       "      <td>11</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0000909</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0001962</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0001389</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0002386</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0001059</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>139 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Aethiopicum  Candidum  Macrocarpon  Muricatum 1  Muricatum 2  \\\n",
       "OG0002827            0         2            2            2            2   \n",
       "OG0005089            0         0            0            0            0   \n",
       "OG0001699            2         2            0            3            3   \n",
       "OG0000192            6         6            5            6            6   \n",
       "OG0000040           10        14           16           13            9   \n",
       "...                ...       ...          ...          ...          ...   \n",
       "OG0000909            3         3            3            2            3   \n",
       "OG0001962            2         2            2            2            2   \n",
       "OG0001389            2         2            2            2            3   \n",
       "OG0002386            2         2            2            2            2   \n",
       "OG0001059            3         3            3            3            3   \n",
       "\n",
       "           Prinophyllum  Quitoense  \n",
       "OG0002827             2          2  \n",
       "OG0005089             0          0  \n",
       "OG0001699             2          2  \n",
       "OG0000192             5          6  \n",
       "OG0000040            11         13  \n",
       "...                 ...        ...  \n",
       "OG0000909             3          4  \n",
       "OG0001962             2          3  \n",
       "OG0001389             2          2  \n",
       "OG0002386             2          2  \n",
       "OG0001059             3          3  \n",
       "\n",
       "[139 rows x 7 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ortho_groups_and_number_of_genes_present_in_each_species #### OG0002827\t is very cool! Lost in Aethiopicum, OG0005089 is lost in everything but tomato"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Aethiopicum</th>\n",
       "      <th>Candidum</th>\n",
       "      <th>Macrocarpon</th>\n",
       "      <th>Muricatum 1</th>\n",
       "      <th>Muricatum 2</th>\n",
       "      <th>Prinophyllum</th>\n",
       "      <th>Quitoense</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>OG0002827</th>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0005089</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0001699</th>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0000192</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0000040</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0000909</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0001962</th>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0001389</th>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0002386</th>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0001059</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>139 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Aethiopicum  Candidum  Macrocarpon  Muricatum 1  Muricatum 2  \\\n",
       "OG0002827        False      True         True         True         True   \n",
       "OG0005089        False     False        False        False        False   \n",
       "OG0001699         True      True        False        False        False   \n",
       "OG0000192        False     False        False        False        False   \n",
       "OG0000040        False     False        False        False        False   \n",
       "...                ...       ...          ...          ...          ...   \n",
       "OG0000909        False     False        False         True        False   \n",
       "OG0001962         True      True         True         True         True   \n",
       "OG0001389         True      True         True         True        False   \n",
       "OG0002386         True      True         True         True         True   \n",
       "OG0001059        False     False        False        False        False   \n",
       "\n",
       "           Prinophyllum  Quitoense  \n",
       "OG0002827          True       True  \n",
       "OG0005089         False      False  \n",
       "OG0001699          True       True  \n",
       "OG0000192         False      False  \n",
       "OG0000040         False      False  \n",
       "...                 ...        ...  \n",
       "OG0000909         False      False  \n",
       "OG0001962          True      False  \n",
       "OG0001389          True       True  \n",
       "OG0002386          True       True  \n",
       "OG0001059         False      False  \n",
       "\n",
       "[139 rows x 7 columns]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ortho_groups_and_number_of_genes_present_in_each_species ==2 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "species_rna_locations ={'Aethiopicum':'/data/passala/Collaborator_Data/Zach_Pan_solanum_Data/Initial_species_bulk_seq/Solanum_Aetheopicum/fpkm_normalized_combined_samples.tsv',\n",
    "    'Candidum':'/data/passala/Collaborator_Data/Zach_Pan_solanum_Data/Initial_species_bulk_seq/Solanum_Candidum/fpkm_normalized_combined_samples.tsv',\n",
    "    'Macrocarpon':'/data/passala/Collaborator_Data/Zach_Pan_solanum_Data/Initial_species_bulk_seq/Solanum_Macrocarpon/fpkm_normalized_combined_samples.tsv',\n",
    "    'Muricatum 1':'/data/passala/Collaborator_Data/Zach_Pan_solanum_Data/Initial_species_bulk_seq/Solanum_muricatum/haplotype_1/fpkm_normalized_combined_samples.tsv',\n",
    "    'Muricatum 2':'/data/passala/Collaborator_Data/Zach_Pan_solanum_Data/Initial_species_bulk_seq/Solanum_muricatum/haplotype_2/fpkm_normalized_combined_samples.tsv',\n",
    "    'Prinophyllum':'/data/passala/Collaborator_Data/Zach_Pan_solanum_Data/Initial_species_bulk_seq/Solanum_Prinophyllum/fpkm_normalized_combined_samples.tsv',\n",
    "    'Quitoense':'/data/passala/Collaborator_Data/Zach_Pan_solanum_Data/Initial_species_bulk_seq/Solanum_Quitoense/fpkm_normalized_combined_samples.tsv'}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "rna_normalized_data = {}\n",
    "for species in species_rna_locations.keys():\n",
    "    current_rna_location = species_rna_locations[species]\n",
    "    current_sample = pd.read_csv(current_rna_location, sep = '\\t', index_col= 0)\n",
    "    current_sample.index = current_sample.index+\".1\"\n",
    "    current_sample['Mean'] = current_sample.mean(axis =1)\n",
    "\n",
    "    rna_normalized_data[species] = current_sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "q_Data = rna_normalized_data['Quitoense']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Rep 1</th>\n",
       "      <th>Rep 2</th>\n",
       "      <th>Mean</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FEATURE_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Solqui2.Cg000010.1</th>\n",
       "      <td>0.154825</td>\n",
       "      <td>0.028682</td>\n",
       "      <td>0.091754</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Solqui2.Cg000020.1</th>\n",
       "      <td>0.088258</td>\n",
       "      <td>0.122627</td>\n",
       "      <td>0.105443</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Solqui2.Cg000050.1</th>\n",
       "      <td>0.186422</td>\n",
       "      <td>0.621646</td>\n",
       "      <td>0.404034</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Solqui2.Cg000060.1</th>\n",
       "      <td>0.101622</td>\n",
       "      <td>0.169435</td>\n",
       "      <td>0.135528</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Solqui2.Cg000070.1</th>\n",
       "      <td>0.537335</td>\n",
       "      <td>0.656994</td>\n",
       "      <td>0.597164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Solqui2.ctg00000691g000010.1</th>\n",
       "      <td>0.023048</td>\n",
       "      <td>0.076856</td>\n",
       "      <td>0.049952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ncRNA_Solqui2.ctg00000739g003721.1</th>\n",
       "      <td>4.515794</td>\n",
       "      <td>0.721979</td>\n",
       "      <td>2.618887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ncRNA_Solqui2.ctg00000739g003731.1</th>\n",
       "      <td>0.056387</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.028193</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ncRNA_Solqui2.ctg00000739g003741.1</th>\n",
       "      <td>0.630536</td>\n",
       "      <td>0.107825</td>\n",
       "      <td>0.369181</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Solqui2.ctg00000760g000010.1</th>\n",
       "      <td>1.124269</td>\n",
       "      <td>0.052069</td>\n",
       "      <td>0.588169</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26455 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       Rep 1     Rep 2      Mean\n",
       "FEATURE_ID                                                      \n",
       "Solqui2.Cg000010.1                  0.154825  0.028682  0.091754\n",
       "Solqui2.Cg000020.1                  0.088258  0.122627  0.105443\n",
       "Solqui2.Cg000050.1                  0.186422  0.621646  0.404034\n",
       "Solqui2.Cg000060.1                  0.101622  0.169435  0.135528\n",
       "Solqui2.Cg000070.1                  0.537335  0.656994  0.597164\n",
       "...                                      ...       ...       ...\n",
       "Solqui2.ctg00000691g000010.1        0.023048  0.076856  0.049952\n",
       "ncRNA_Solqui2.ctg00000739g003721.1  4.515794  0.721979  2.618887\n",
       "ncRNA_Solqui2.ctg00000739g003731.1  0.056387  0.000000  0.028193\n",
       "ncRNA_Solqui2.ctg00000739g003741.1  0.630536  0.107825  0.369181\n",
       "Solqui2.ctg00000760g000010.1        1.124269  0.052069  0.588169\n",
       "\n",
       "[26455 rows x 3 columns]"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "q_Data.loc[q_Data['Mean']!= 0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Orthogroup</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Rep 1</th>\n",
       "      <th>Rep 2</th>\n",
       "      <th>Mean</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5545</th>\n",
       "      <td>OG0001059</td>\n",
       "      <td>Solqui2.03g005640.1</td>\n",
       "      <td>41.638838</td>\n",
       "      <td>46.330493</td>\n",
       "      <td>43.984665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5546</th>\n",
       "      <td>OG0001059</td>\n",
       "      <td>Solqui2.07g021260.1</td>\n",
       "      <td>129.885545</td>\n",
       "      <td>161.518000</td>\n",
       "      <td>145.701772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5547</th>\n",
       "      <td>OG0001059</td>\n",
       "      <td>Solqui2.10g003040.1</td>\n",
       "      <td>91.200498</td>\n",
       "      <td>80.056813</td>\n",
       "      <td>85.628656</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Orthogroup                 Gene       Rep 1       Rep 2        Mean\n",
       "5545  OG0001059  Solqui2.03g005640.1   41.638838   46.330493   43.984665\n",
       "5546  OG0001059  Solqui2.07g021260.1  129.885545  161.518000  145.701772\n",
       "5547  OG0001059  Solqui2.10g003040.1   91.200498   80.056813   85.628656"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_in_orthogroup.merge(right = q_Data, left_on = 'Gene', right_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "ortho_groups_and_number_of_genes_present_in_each_species = pd.DataFrame(index = list_of_orthogroups_with_fc_thresholded_tomato)\n",
    "for species in ortho_group_dictionary_of_gene_groups.keys():\n",
    "    current_list_of_genes = ortho_group_dictionary_of_gene_groups[species]\n",
    "    all_orthogroups_number_present = []\n",
    "    rna_data_for_species = rna_normalized_data[species]\n",
    "    for orthogroup in list_of_orthogroups_with_fc_thresholded_tomato:\n",
    "        genes_in_orthogroup = current_list_of_genes.loc[current_list_of_genes['Orthogroup'] == orthogroup]\n",
    "        merged_genes_with_reads = genes_in_orthogroup.merge(right = rna_data_for_species, left_on = 'Gene', right_index=True)\n",
    "        merged_genes_no_zeros = merged_genes_with_reads.loc[merged_genes_with_reads['Mean']!= 0]\n",
    "        numb_of_that_gene_in_species = len(merged_genes_no_zeros)\n",
    "        all_orthogroups_number_present.append(numb_of_that_gene_in_species)\n",
    "    ortho_groups_and_number_of_genes_present_in_each_species[species] = all_orthogroups_number_present"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#### 1 or zero represents loss vs tomato \n",
    "#### Find lost "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "OG0002827     True\n",
       "OG0005089     True\n",
       "OG0001699    False\n",
       "OG0000192    False\n",
       "OG0000040    False\n",
       "             ...  \n",
       "OG0000909    False\n",
       "OG0001962     True\n",
       "OG0001389     True\n",
       "OG0002386     True\n",
       "OG0001059    False\n",
       "Length: 139, dtype: bool"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(ortho_groups_and_number_of_genes_present_in_each_species <3).all(axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Aethiopicum</th>\n",
       "      <th>Candidum</th>\n",
       "      <th>Macrocarpon</th>\n",
       "      <th>Muricatum 1</th>\n",
       "      <th>Muricatum 2</th>\n",
       "      <th>Prinophyllum</th>\n",
       "      <th>Quitoense</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>OG0002827</th>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0005089</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0005408</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0023112</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0002577</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0003580</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0004370</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0001962</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0001389</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OG0002386</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>75 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Aethiopicum  Candidum  Macrocarpon  Muricatum 1  Muricatum 2  \\\n",
       "OG0002827            0         2            2            2            2   \n",
       "OG0005089            0         0            0            0            0   \n",
       "OG0005408            0         0            0            0            0   \n",
       "OG0023112            0         0            0            0            0   \n",
       "OG0002577            2         2            2            2            2   \n",
       "...                ...       ...          ...          ...          ...   \n",
       "OG0003580            0         0            2            2            2   \n",
       "OG0004370            0         0            0            0            0   \n",
       "OG0001962            2         2            2            2            2   \n",
       "OG0001389            1         1            2            1            1   \n",
       "OG0002386            2         2            2            2            2   \n",
       "\n",
       "           Prinophyllum  Quitoense  \n",
       "OG0002827             2          2  \n",
       "OG0005089             0          0  \n",
       "OG0005408             0          0  \n",
       "OG0023112             0          0  \n",
       "OG0002577             2          2  \n",
       "...                 ...        ...  \n",
       "OG0003580             2          0  \n",
       "OG0004370             0          0  \n",
       "OG0001962             2          1  \n",
       "OG0001389             2          1  \n",
       "OG0002386             2          2  \n",
       "\n",
       "[75 rows x 7 columns]"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ortho_groups_and_number_of_genes_present_in_each_species.loc[(ortho_groups_and_number_of_genes_present_in_each_species <3).all(axis = 1)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(0.0, 1000.0)"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAlYAAAG2CAYAAAC9CcgAAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8qNh9FAAAACXBIWXMAAA9hAAAPYQGoP6dpAAAqkUlEQVR4nO3de3SU5YHH8d+QhCGkyQgJZDISMGxTAYOCgVKQLViuK8hh3VNALoWKCuUaLgUptiJdEqEV2AriYi1YKI3tFrrsLlXCxQgNCg1EuamrUu4xroYJlJhg8uwfHt7TIQFJeJKZSb6fc+bUed9nZp53nsPJt+9M3riMMUYAAAC4ZU2CPQEAAICGgrACAACwhLACAACwhLACAACwhLACAACwhLACAACwhLACAACwhLACAACwhLACAACwhLACAACwJKhh9cYbb+jBBx+Uz+eTy+XSH//4x4D9xhgtWrRIPp9P0dHR6tu3r44ePRowpqysTNOnT1dCQoJiYmI0bNgwnTlzJmBMcXGxxo0bJ4/HI4/Ho3HjxunChQt1fHQAAKCxCWpY/e1vf9M999yjVatWVbt/2bJlWr58uVatWqUDBw7I6/VqwIABunjxojMmIyNDW7ZsUXZ2tvbu3atLly5p6NChqqiocMaMHj1aBQUFevXVV/Xqq6+qoKBA48aNq/PjAwAAjYsrVP4Is8vl0pYtWzR8+HBJX56t8vl8ysjI0Pz58yV9eXYqMTFRS5cu1aRJk+T3+9WqVStt2LBBI0eOlCSdO3dOycnJ2rZtmwYNGqTjx4+rU6dOevPNN9WjRw9J0ptvvqmePXvq3Xff1Z133hmU4wUAAA1PZLAncD0nTpxQYWGhBg4c6Gxzu93q06eP8vLyNGnSJOXn5+vKlSsBY3w+n9LS0pSXl6dBgwZp37598ng8TlRJ0re+9S15PB7l5eVdN6zKyspUVlbm3K+srNRnn32m+Ph4uVyuOjhiAABgmzFGFy9elM/nU5Mmdf9BXciGVWFhoSQpMTExYHtiYqJOnjzpjGnatKlatGhRZczVxxcWFqp169ZVnr9169bOmOpkZWXp6aefvqVjAAAAoeH06dNq06ZNnb9OyIbVVdeeHTLGfOUZo2vHVDf+q55nwYIFmj17tnPf7/erbdu2On36tOLi4m52+gAAIIhKSkqUnJys2NjYenm9kA0rr9cr6cszTklJSc72oqIi5yyW1+tVeXm5iouLA85aFRUVqVevXs6Yjz/+uMrzf/LJJ1XOhv09t9stt9tdZXtcXBxhBQBAmKmvr/GE7HWsUlJS5PV6lZOT42wrLy9Xbm6uE03p6emKiooKGHP+/HkdOXLEGdOzZ0/5/X7t37/fGfPWW2/J7/c7YwAAAGwI6hmrS5cu6YMPPnDunzhxQgUFBWrZsqXatm2rjIwMZWZmKjU1VampqcrMzFTz5s01evRoSZLH49HEiRM1Z84cxcfHq2XLlpo7d646d+6s/v37S5I6duyowYMH67HHHtO///u/S5Ief/xxDR06lN8IBAAAVgU1rP7yl7/o/vvvd+5f/U7T+PHjtX79es2bN0+lpaWaMmWKiouL1aNHD23fvj3gc9IVK1YoMjJSI0aMUGlpqfr166f169crIiLCGfOb3/xGM2bMcH57cNiwYde9dhYAAEBthcx1rEJdSUmJPB6P/H4/37ECACBM1PfP75D9jhUAAEC4IawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsCemw+uKLL/Tkk08qJSVF0dHRat++vRYvXqzKykpnjDFGixYtks/nU3R0tPr27aujR48GPE9ZWZmmT5+uhIQExcTEaNiwYTpz5kx9Hw4AAGjgQjqsli5dqhdeeEGrVq3S8ePHtWzZMv3sZz/Tc88954xZtmyZli9frlWrVunAgQPyer0aMGCALl686IzJyMjQli1blJ2drb179+rSpUsaOnSoKioqgnFYAACggXIZY0ywJ3E9Q4cOVWJiol566SVn27/8y7+oefPm2rBhg4wx8vl8ysjI0Pz58yV9eXYqMTFRS5cu1aRJk+T3+9WqVStt2LBBI0eOlCSdO3dOycnJ2rZtmwYNGnRTcykpKZHH45Hf71dcXJz9gwUAANbV98/vkD5j1bt3b+3cuVPvv/++JOntt9/W3r179cADD0iSTpw4ocLCQg0cONB5jNvtVp8+fZSXlydJys/P15UrVwLG+Hw+paWlOWOqU1ZWppKSkoAbAADAjUQGewI3Mn/+fPn9fnXo0EERERGqqKjQkiVL9PDDD0uSCgsLJUmJiYkBj0tMTNTJkyedMU2bNlWLFi2qjLn6+OpkZWXp6aeftnk4AACggQvpM1avvPKKNm7cqE2bNungwYN6+eWX9fOf/1wvv/xywDiXyxVw3xhTZdu1vmrMggUL5Pf7ndvp06drfyAAAKBRCOkzVj/84Q/1xBNPaNSoUZKkzp076+TJk8rKytL48ePl9XolfXlWKikpyXlcUVGRcxbL6/WqvLxcxcXFAWetioqK1KtXr+u+ttvtltvtrovDAgAADVRIn7G6fPmymjQJnGJERIRzuYWUlBR5vV7l5OQ4+8vLy5Wbm+tEU3p6uqKiogLGnD9/XkeOHLlhWAEAANRUSJ+xevDBB7VkyRK1bdtWd911lw4dOqTly5frkUcekfTlR4AZGRnKzMxUamqqUlNTlZmZqebNm2v06NGSJI/Ho4kTJ2rOnDmKj49Xy5YtNXfuXHXu3Fn9+/cP5uEBAIAGJqTD6rnnntOPf/xjTZkyRUVFRfL5fJo0aZJ+8pOfOGPmzZun0tJSTZkyRcXFxerRo4e2b9+u2NhYZ8yKFSsUGRmpESNGqLS0VP369dP69esVERERjMMCAAANVEhfxyqUcB0rAADCD9exAgAACFOEFQAAgCWEFQAAgCWEFQAAgCWEFQAAgCWEFQAAgCWEFQAAgCWEFQAAgCWEFQAAgCWEFQAAgCWEFQAAgCWEFQAAgCWEFQAAgCWEVQ19f93+YE8BAACEKMIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsIKAADAEsKqFkat3RfsKQAAgBBEWAEAAFhCWAEAAFhCWAEAAFhCWAEAAFhCWAEAAFhCWAEAAFhCWAEAAFgS8mF19uxZjR07VvHx8WrevLm6dOmi/Px8Z78xRosWLZLP51N0dLT69u2ro0ePBjxHWVmZpk+froSEBMXExGjYsGE6c+ZMfR8KAABo4EI6rIqLi3XfffcpKipKf/rTn3Ts2DE9++yzuu2225wxy5Yt0/Lly7Vq1SodOHBAXq9XAwYM0MWLF50xGRkZ2rJli7Kzs7V3715dunRJQ4cOVUVFRRCOCgAANFQuY4wJ9iSu54knntCf//xn7dmzp9r9xhj5fD5lZGRo/vz5kr48O5WYmKilS5dq0qRJ8vv9atWqlTZs2KCRI0dKks6dO6fk5GRt27ZNgwYNuqm5lJSUyOPx6KGVOYqKjlH24z3tHCQAAKgzV39++/1+xcXF1fnrhfQZq61bt6pbt2767ne/q9atW6tr16568cUXnf0nTpxQYWGhBg4c6Gxzu93q06eP8vLyJEn5+fm6cuVKwBifz6e0tDRnTHXKyspUUlIScAMAALiRkA6rjz76SGvWrFFqaqpee+01TZ48WTNmzNCvf/1rSVJhYaEkKTExMeBxiYmJzr7CwkI1bdpULVq0uO6Y6mRlZcnj8Ti35ORkm4cGAAAaoJAOq8rKSt17773KzMxU165dNWnSJD322GNas2ZNwDiXyxVw3xhTZdu1vmrMggUL5Pf7ndvp06drfyAAAKBRCOmwSkpKUqdOnQK2dezYUadOnZIkeb1eSapy5qmoqMg5i+X1elVeXq7i4uLrjqmO2+1WXFxcwA0AAOBGQjqs7rvvPr333nsB295//321a9dOkpSSkiKv16ucnBxnf3l5uXJzc9WrVy9JUnp6uqKiogLGnD9/XkeOHHHGAAAA2BAZ7AncyKxZs9SrVy9lZmZqxIgR2r9/v9auXau1a9dK+vIjwIyMDGVmZio1NVWpqanKzMxU8+bNNXr0aEmSx+PRxIkTNWfOHMXHx6tly5aaO3euOnfurP79+wfz8AAAQAMT0mHVvXt3bdmyRQsWLNDixYuVkpKilStXasyYMc6YefPmqbS0VFOmTFFxcbF69Oih7du3KzY21hmzYsUKRUZGasSIESotLVW/fv20fv16RUREBOOwAABAAxXS17EKJVzHCgCA8MN1rAAAAMIUYQUAAGAJYQUAAGAJYQUAAGAJYQUAAGAJYQUAAGAJYQUAAGAJYQUAAGAJYQUAAGAJYQUAAGAJYQUAAGAJYQUAAGBJrcKqffv2+vTTT6tsv3Dhgtq3b3/LkwIAAAhHtQqrv/71r6qoqKiyvaysTGfPnr3lSQEAAISjyJoM3rp1q/Pfr732mjwej3O/oqJCO3fu1B133GFtcgAAAOGkRmE1fPhwSZLL5dL48eMD9kVFRemOO+7Qs88+a21yAAAA4aRGYVVZWSlJSklJ0YEDB5SQkFAnkwIAAAhHNQqrq06cOGF7HgAAAGGvVmElSTt37tTOnTtVVFTknMm66le/+tUtTwwAACDc1Cqsnn76aS1evFjdunVTUlKSXC6X7XkBAACEnVqF1QsvvKD169dr3LhxtucDAAAQtmp1Havy8nL16tXL9lwAAADCWq3C6tFHH9WmTZtszwUAACCs1eqjwM8//1xr167Vjh07dPfddysqKipg//Lly61MLpSNWrtP2Y/3DPY0AABACKlVWL3zzjvq0qWLJOnIkSMB+/giOwAAaKxqFVa7d++2PQ8AAICwV6vvWAEAAKCqWp2xuv/++2/4kd+uXbtqPSEAAIBwVauwuvr9qquuXLmigoICHTlypMofZwYAAGgsahVWK1asqHb7okWLdOnSpVuaEAAAQLiy+h2rsWPH8ncCAQBAo2U1rPbt26dmzZrZfEoAAICwUauPAh966KGA+8YYnT9/Xn/5y1/04x//2MrEwgEXCQUAAH+vVmHl8XgC7jdp0kR33nmnFi9erIEDB1qZGAAAQLipVVitW7fO9jwAAADCXq3C6qr8/HwdP35cLpdLnTp1UteuXW3NCwAAIOzUKqyKioo0atQovf7667rttttkjJHf79f999+v7OxstWrVyvY8AQAAQl6tfitw+vTpKikp0dGjR/XZZ5+puLhYR44cUUlJiWbMmGF7jgAAAGGhVmesXn31Ve3YsUMdO3Z0tnXq1EmrV6/my+sAAKDRqtUZq8rKSkVFRVXZHhUVpcrKylueFAAAQDiqVVh95zvf0cyZM3Xu3Dln29mzZzVr1iz169fP2uQAAADCSa3CatWqVbp48aLuuOMO/cM//IO+/vWvKyUlRRcvXtRzzz1ne44AAABhoVbfsUpOTtbBgweVk5Ojd999V8YYderUSf3797c9v5DH1dcBAMBVNTpjtWvXLnXq1EklJSWSpAEDBmj69OmaMWOGunfvrrvuukt79uypk4kCAACEuhqF1cqVK/XYY48pLi6uyj6Px6NJkyZp+fLl1iYHAAAQTmoUVm+//bYGDx583f0DBw5Ufn7+LU8KAAAgHNUorD7++ONqL7NwVWRkpD755JNbnhQAAEA4qlFY3X777Tp8+PB197/zzjtKSkq65UkBAACEoxqF1QMPPKCf/OQn+vzzz6vsKy0t1VNPPaWhQ4damxwAAEA4qdHlFp588klt3rxZ3/jGNzRt2jTdeeedcrlcOn78uFavXq2KigotXLiwruYKAAAQ0moUVomJicrLy9MPfvADLViwQMYYSZLL5dKgQYP0/PPPKzExsU4mCgAAEOpqfIHQdu3aadu2bSouLtYHH3wgY4xSU1PVokWLupgfAABA2KjVldclqUWLFurevbvNuQAAAIS1Wv2tQAAAAFRFWAEAAFhCWAEAAFhCWAEAAFhCWAEAAFhCWAEAAFhCWAEAAFhCWAEAAFhCWFkwau2+YE8BAACEAMIKAADAkrAKq6ysLLlcLmVkZDjbjDFatGiRfD6foqOj1bdvXx09ejTgcWVlZZo+fboSEhIUExOjYcOG6cyZM/U8ewAA0NCFTVgdOHBAa9eu1d133x2wfdmyZVq+fLlWrVqlAwcOyOv1asCAAbp48aIzJiMjQ1u2bFF2drb27t2rS5cuaejQoaqoqKjvwwAAAA1YWITVpUuXNGbMGL344otq0aKFs90Yo5UrV2rhwoV66KGHlJaWppdfflmXL1/Wpk2bJEl+v18vvfSSnn32WfXv319du3bVxo0bdfjwYe3YsSNYhwQAABqgsAirqVOnasiQIerfv3/A9hMnTqiwsFADBw50trndbvXp00d5eXmSpPz8fF25ciVgjM/nU1pamjOmOmVlZSopKQm4AQAA3EhksCfwVbKzs3Xw4EEdOHCgyr7CwkJJUmJiYsD2xMREnTx50hnTtGnTgDNdV8dcfXx1srKy9PTTT9/q9AEAQCMS0mesTp8+rZkzZ2rjxo1q1qzZdce5XK6A+8aYKtuu9VVjFixYIL/f79xOnz5ds8kDAIBGJ6TDKj8/X0VFRUpPT1dkZKQiIyOVm5urX/ziF4qMjHTOVF175qmoqMjZ5/V6VV5eruLi4uuOqY7b7VZcXFzADQAA4EZCOqz69eunw4cPq6CgwLl169ZNY8aMUUFBgdq3by+v16ucnBznMeXl5crNzVWvXr0kSenp6YqKigoYc/78eR05csQZAwAAYENIf8cqNjZWaWlpAdtiYmIUHx/vbM/IyFBmZqZSU1OVmpqqzMxMNW/eXKNHj5YkeTweTZw4UXPmzFF8fLxatmypuXPnqnPnzlW+DA8AAHArQjqsbsa8efNUWlqqKVOmqLi4WD169ND27dsVGxvrjFmxYoUiIyM1YsQIlZaWql+/flq/fr0iIiKCOHMAANDQuIwxJtiTCAclJSXyeDx6aGWOoqJjquzPfrxnEGYFAABu5OrPb7/fXy/flw7p71gBAACEE8IKAADAEsIKAADAEsIKAADAEsLKklFr9wV7CgAAIMgIKwAAAEsIKwAAAEsIKwAAAEsIKwAAAEsIKwAAAEsIKwAAAEsIKwAAAEsIKwAAAEsIKwAAAEsIKwAAAEsIKwAAAEsIKwAAAEsIKwAAAEsIKwAAAEsIKwAAAEsIKwAAAEsIKwAAAEsIK4tGrd0X7CkAAIAgIqwAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawsG7V2X7CnAAAAgoSwAgAAsISwqgOctQIAoHEirAAAACwhrAAAACwhrAAAACwhrAAAACwhrAAAACwhrAAAACwhrAAAACwhrAAAACwhrAAAACwhrAAAACwhrOoIf9YGAIDGJ6TDKisrS927d1dsbKxat26t4cOH67333gsYY4zRokWL5PP5FB0drb59++ro0aMBY8rKyjR9+nQlJCQoJiZGw4YN05kzZ+rzUAAAQCMQ0mGVm5urqVOn6s0331ROTo6++OILDRw4UH/729+cMcuWLdPy5cu1atUqHThwQF6vVwMGDNDFixedMRkZGdqyZYuys7O1d+9eXbp0SUOHDlVFRUUwDgsAADRQLmOMCfYkbtYnn3yi1q1bKzc3V9/+9rdljJHP51NGRobmz58v6cuzU4mJiVq6dKkmTZokv9+vVq1aacOGDRo5cqQk6dy5c0pOTta2bds0aNCgm3rtkpISeTwePbQyR1HRMTf1mOzHe9buQAEAgBVXf377/X7FxcXV+euF9Bmra/n9fklSy5YtJUknTpxQYWGhBg4c6Ixxu93q06eP8vLyJEn5+fm6cuVKwBifz6e0tDRnTHXKyspUUlIScAMAALiRsAkrY4xmz56t3r17Ky0tTZJUWFgoSUpMTAwYm5iY6OwrLCxU06ZN1aJFi+uOqU5WVpY8Ho9zS05Otnk4AACgAQqbsJo2bZreeecd/fa3v62yz+VyBdw3xlTZdq2vGrNgwQL5/X7ndvr06dpNHAAANBphEVbTp0/X1q1btXv3brVp08bZ7vV6JanKmaeioiLnLJbX61V5ebmKi4uvO6Y6brdbcXFxATcAAIAbCemwMsZo2rRp2rx5s3bt2qWUlJSA/SkpKfJ6vcrJyXG2lZeXKzc3V7169ZIkpaenKyoqKmDM+fPndeTIEWcMAACADZHBnsCNTJ06VZs2bdJ//ud/KjY21jkz5fF4FB0dLZfLpYyMDGVmZio1NVWpqanKzMxU8+bNNXr0aGfsxIkTNWfOHMXHx6tly5aaO3euOnfurP79+wfz8AAAQAMT0mG1Zs0aSVLfvn0Dtq9bt04TJkyQJM2bN0+lpaWaMmWKiouL1aNHD23fvl2xsbHO+BUrVigyMlIjRoxQaWmp+vXrp/Xr1ysiIqK+DgUAADQCYXUdq2DiOlYAAIQfrmMFAAAQpggrAAAASwgrAAAASwirOjRq7b5gTwEAANQjwgoAAMASwgoAAMASwqoe8JEgAACNA2EFAABgCWEFAABgCWFVx/gYEACAxoOwAgAAsISwAgAAsISwAgAAsISwAgAAsISwAgAAsISwAgAAsISwAgAAsISwAgAAsISwAgAAsISwAgAAsISwAgAAsISwAgAAsISwAgAAsISwqiej1u4L9hQAAEAdI6wAAAAsIawAAAAsIazqGR8JAgDQcBFW9YioAgCgYSOsAAAALCGsAAAALCGsgoSPBQEAaHgIqyAgqgAAaJgIKwAAAEsIKwAAAEsIKwAAAEsIKwAAAEsIKwAAAEsIKwAAAEsIqyDisgsAADQshBUAAIAlhBUAAIAlhBUAAIAlhFUI4LtWAAA0DIRVkBFVAAA0HIQVAACAJYRViODMFQAA4Y+wCiHEFQAA4Y2wAgAAsISwAgAAsISwCjF8HAgAQPgirAAAACwhrEIUZ64AAAg/hFUIIqoAAAhPhFUIu1FgEV8AAIQewgoAAMASwirMcKYKAIDQRViFgVFr9xFUAACEAcIqxFUXVEQWAAChibAKI0QWAAChjbAKY0QVAAChhbBqQAgtAACCi7BqAK79cnt19wEAQN0jrBqYGwXWrYwBAABfrVGF1fPPP6+UlBQ1a9ZM6enp2rNnT7CnVG9uNo6uN+7q9mv/90aPJ8gAAI1NowmrV155RRkZGVq4cKEOHTqkf/zHf9Q//dM/6dSpU8GeWr2p6UeENxNZtxJPhBcAoKFpNGG1fPlyTZw4UY8++qg6duyolStXKjk5WWvWrAn21ILi2qj5+0i6XnzdbIhV91xXt90o1m7mY8lbfcyNxtzq5SxqOvfavEa4a0zHCqBxchljTLAnUdfKy8vVvHlz/f73v9c///M/O9tnzpypgoIC5ebmVnlMWVmZysrKnPt+v19t27bVg8/8UVHNYupl3qjeuu9/U99ft79Ox93MY6835ur2a/+3pvOo7r4k5zmvut5rVTf+7/fdzPi/d73XvdG467329eZd3fNV99jazO16x1vdXG40h5t57PX++0b3v2oON3rOm3neG703N3K917r2fbjRc9/sa9Z0bnXtZuZjY851cdy3uu63+nq2x9+KkpISJScn68KFC/J4PHX/gqYROHv2rJFk/vznPwdsX7JkifnGN75R7WOeeuopI4kbN27cuHHj1gBuH374YX0kh4lUI+JyuQLuG2OqbLtqwYIFmj17tnP/woULateunU6dOlU/xYvruvr/Pk6fPq24uLhgT6dRYy1CB2sRWliP0HH1E6eWLVvWy+s1irBKSEhQRESECgsLA7YXFRUpMTGx2se43W653e4q2z0eD/9IQkRcXBxrESJYi9DBWoQW1iN0NGlSP18rbxRfXm/atKnS09OVk5MTsD0nJ0e9evUK0qwAAEBD0yjOWEnS7NmzNW7cOHXr1k09e/bU2rVrderUKU2ePDnYUwMAAA1EowmrkSNH6tNPP9XixYt1/vx5paWladu2bWrXrt1NPd7tduupp56q9uNB1C/WInSwFqGDtQgtrEfoqO+1aBSXWwAAAKgPjeI7VgAAAPWBsAIAALCEsAIAALCEsAIAALCEsLoJzz//vFJSUtSsWTOlp6drz549wZ5Sg5OVlaXu3bsrNjZWrVu31vDhw/Xee+8FjDHGaNGiRfL5fIqOjlbfvn119OjRgDFlZWWaPn26EhISFBMTo2HDhunMmTP1eSgNTlZWllwulzIyMpxtrEX9OXv2rMaOHav4+Hg1b95cXbp0UX5+vrOftagfX3zxhZ588kmlpKQoOjpa7du31+LFi1VZWemMYS3qzhtvvKEHH3xQPp9PLpdLf/zjHwP223rvi4uLNW7cOHk8Hnk8Ho0bN04XLlyo2WTr5Q/nhLHs7GwTFRVlXnzxRXPs2DEzc+ZMExMTY06ePBnsqTUogwYNMuvWrTNHjhwxBQUFZsiQIaZt27bm0qVLzphnnnnGxMbGmj/84Q/m8OHDZuTIkSYpKcmUlJQ4YyZPnmxuv/12k5OTYw4ePGjuv/9+c88995gvvvgiGIcV9vbv32/uuOMOc/fdd5uZM2c621mL+vHZZ5+Zdu3amQkTJpi33nrLnDhxwuzYscN88MEHzhjWon7867/+q4mPjzf//d//bU6cOGF+//vfm6997Wtm5cqVzhjWou5s27bNLFy40PzhD38wksyWLVsC9tt67wcPHmzS0tJMXl6eycvLM2lpaWbo0KE1mith9RW++c1vmsmTJwds69Chg3niiSeCNKPGoaioyEgyubm5xhhjKisrjdfrNc8884wz5vPPPzcej8e88MILxhhjLly4YKKiokx2drYz5uzZs6ZJkybm1Vdfrd8DaAAuXrxoUlNTTU5OjunTp48TVqxF/Zk/f77p3bv3dfezFvVnyJAh5pFHHgnY9tBDD5mxY8caY1iL+nRtWNl6748dO2YkmTfffNMZs2/fPiPJvPvuuzc9Pz4KvIHy8nLl5+dr4MCBAdsHDhyovLy8IM2qcfD7/ZLk/NHMEydOqLCwMGAt3G63+vTp46xFfn6+rly5EjDG5/MpLS2N9aqFqVOnasiQIerfv3/Adtai/mzdulXdunXTd7/7XbVu3Vpdu3bViy++6OxnLepP7969tXPnTr3//vuSpLffflt79+7VAw88IIm1CCZb7/2+ffvk8XjUo0cPZ8y3vvUteTyeGq1Po7nyem383//9nyoqKqr8oebExMQqf9AZ9hhjNHv2bPXu3VtpaWmS5Lzf1a3FyZMnnTFNmzZVixYtqoxhvWomOztbBw8e1IEDB6rsYy3qz0cffaQ1a9Zo9uzZ+tGPfqT9+/drxowZcrvd+t73vsda1KP58+fL7/erQ4cOioiIUEVFhZYsWaKHH35YEv8ugsnWe19YWKjWrVtXef7WrVvXaH0Iq5vgcrkC7htjqmyDPdOmTdM777yjvXv3VtlXm7VgvWrm9OnTmjlzprZv365mzZpddxxrUfcqKyvVrVs3ZWZmSpK6du2qo0ePas2aNfre977njGMt6t4rr7yijRs3atOmTbrrrrtUUFCgjIwM+Xw+jR8/3hnHWgSPjfe+uvE1XR8+CryBhIQERUREVCnVoqKiKmUMO6ZPn66tW7dq9+7datOmjbPd6/VK0g3Xwuv1qry8XMXFxdcdg6+Wn5+voqIipaenKzIyUpGRkcrNzdUvfvELRUZGOu8la1H3kpKS1KlTp4BtHTt21KlTpyTx76I+/fCHP9QTTzyhUaNGqXPnzho3bpxmzZqlrKwsSaxFMNl6771erz7++OMqz//JJ5/UaH0Iqxto2rSp0tPTlZOTE7A9JydHvXr1CtKsGiZjjKZNm6bNmzdr165dSklJCdifkpIir9cbsBbl5eXKzc111iI9PV1RUVEBY86fP68jR46wXjXQr18/HT58WAUFBc6tW7duGjNmjAoKCtS+fXvWop7cd999VS478v777zt/PJ5/F/Xn8uXLatIk8EdmRESEc7kF1iJ4bL33PXv2lN/v1/79+50xb731lvx+f83W5+a/h984Xb3cwksvvWSOHTtmMjIyTExMjPnrX/8a7Kk1KD/4wQ+Mx+Mxr7/+ujl//rxzu3z5sjPmmWeeMR6Px2zevNkcPnzYPPzww9X+Om2bNm3Mjh07zMGDB813vvMdfpXZgr//rUBjWIv6sn//fhMZGWmWLFli/vd//9f85je/Mc2bNzcbN250xrAW9WP8+PHm9ttvdy63sHnzZpOQkGDmzZvnjGEt6s7FixfNoUOHzKFDh4wks3z5cnPo0CHn0ke23vvBgwebu+++2+zbt8/s27fPdO7cmcst1IXVq1ebdu3amaZNm5p7773XuQQA7JFU7W3dunXOmMrKSvPUU08Zr9dr3G63+fa3v20OHz4c8DylpaVm2rRppmXLliY6OtoMHTrUnDp1qp6PpuG5NqxYi/rzX//1XyYtLc243W7ToUMHs3bt2oD9rEX9KCkpMTNnzjRt27Y1zZo1M+3btzcLFy40ZWVlzhjWou7s3r272p8R48ePN8bYe+8//fRTM2bMGBMbG2tiY2PNmDFjTHFxcY3m6jLGmFqceQMAAMA1+I4VAACAJYQVAACAJYQVAACAJYQVAACAJYQVAACAJYQVAACAJYQVAACAJYQVAACAJYQVgLA2YcIEuVwuTZ48ucq+KVOmyOVyacKECfU/MQCNEmEFIOwlJycrOztbpaWlzrbPP/9cv/3tb9W2bdsgzgxAY0NYAQh79957r9q2bavNmzc72zZv3qzk5GR17drV2WaM0bJly9S+fXtFR0frnnvu0X/8x384+ysqKjRx4kSlpKQoOjpad955p/7t3/4t4LUmTJig4cOH6+c//7mSkpIUHx+vqVOn6sqVK3V/oABCXmSwJwAANnz/+9/XunXrNGbMGEnSr371Kz3yyCN6/fXXnTFPPvmkNm/erDVr1ig1NVVvvPGGxo4dq1atWqlPnz6qrKxUmzZt9Lvf/U4JCQnKy8vT448/rqSkJI0YMcJ5nt27dyspKUm7d+/WBx98oJEjR6pLly567LHH6vuwAYQY/ggzgLA2YcIEXbhwQb/85S/Vpk0bvfvuu3K5XOrQoYNOnz6tRx99VLfddptWr16thIQE7dq1Sz179nQe/+ijj+ry5cvatGlTtc8/depUffzxx86ZrQkTJuj111/Xhx9+qIiICEnSiBEj1KRJE2VnZ9f9AQMIaZyxAtAgJCQkaMiQIXr55ZdljNGQIUOUkJDg7D927Jg+//xzDRgwIOBx5eXlAR8XvvDCC/rlL3+pkydPqrS0VOXl5erSpUvAY+666y4nqiQpKSlJhw8frpsDAxBWCCsADcYjjzyiadOmSZJWr14dsK+yslKS9D//8z+6/fbbA/a53W5J0u9+9zvNmjVLzz77rHr27KnY2Fj97Gc/01tvvRUwPioqKuC+y+Vynh9A40ZYAWgwBg8erPLycknSoEGDAvZ16tRJbrdbp06dUp8+fap9/J49e9SrVy9NmTLF2fbhhx/W3YQBNDiEFYAGIyIiQsePH3f+++/FxsZq7ty5mjVrliorK9W7d2+VlJQoLy9PX/va1zR+/Hh9/etf169//Wu99tprSklJ0YYNG3TgwAGlpKQE43AAhCHCCkCDEhcXd919P/3pT9W6dWtlZWXpo48+0m233aZ7771XP/rRjyRJkydPVkFBgUaOHCmXy6WHH35YU6ZM0Z/+9Kf6mj6AMMdvBQIAAFjCBUIBAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAsIawAAAAs+X81zgCBOPuQhQAAAABJRU5ErkJggg==",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sns.histplot(rna_normalized_data[\"Quitoense\"]['Mean'])\n",
    "plt.xlim((0,1000))\n",
    "plt.ylim((0,1000))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Block for when groups are bigger than 2\n",
    "# list_of_gene_pairs_in_ortho_group = []\n",
    "# for ortho in set(quitoense_long_df['Orthogroup']):\n",
    "#     matched_long_df = quitoense_long_df.loc[quitoense_long_df['Orthogroup'] == ortho]\n",
    "#     all_combinations_in_orthogroup = list(itertools.combinations(matched_long_df['Gene'].to_list(),2))\n",
    "#     list_of_gene_pairs_in_ortho_group.extend(all_combinations_in_orthogroup)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.9.13 ('Single_cell_data')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "530d15c8ba2fc1aa8d54f337a1499597a993cb74b2c7b7d404487a41cb510e6b"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
